League/Repository/ScopeRepository.php

```diff
@@ -83,4 +88,32 @@ public function finalizeScopes(
 
         return $this->scopeConverter->toLeagueArray($event->getScopes());
     }
+
+    /**
+     * @return ScopeModel[]
+     */
+    private function setupScopes(ClientModel $client, array $requestedScopes): array
```

Comment:
@spideyfusion I would suggest to change return type to be iterable

Reply:
Why? Both methods: ClientModel->getScopes, scopeConverter->toDomainArray and current method logic provide clean return type "array"

Reply:
@Derkien You bring up a good point. I have reverted the change.